test/utils/config.go

```diff
@@ -2,8 +2,52 @@ package utils
 
 import "testing"
 
+type Config struct {
+	TestChangePort ConfigChangePort
+}
+
+type ConfigChangePort struct {
+	App                      string
+	DefaultPort              string
+	TestLegacyEnvConfig      bool
+	TestAppConfig            bool
+	TestGlobalConfig         bool
+	TestMixedGlobalAppConfig bool
+}
+
 const serviceWaitTimeout = 60 // seconds
 
+func TestConfig(t *testing.T, snapName string, conf Config) {
+	t.Run("config", func(t *testing.T) {
+		TestChangePort(t, snapName, conf.TestChangePort)
+	})
+}
+
+func TestChangePort(t *testing.T, snapName string, conf ConfigChangePort) {
+	t.Run("change service port", func(t *testing.T) {
+
+		// start once so that default configs get uploaded to the registry
+		service := snapName + "." + conf.App
+		SnapStart(nil, service)
+		WaitServiceOnline(nil, 60, conf.DefaultPort)
+		SnapStop(nil, service)
+
+		if conf.TestLegacyEnvConfig {
+			SetEnvConfig(t, snapName, conf.App, conf.DefaultPort)
+		}
+		if conf.TestAppConfig {
+			SetAppConfig(t, snapName, conf.App, conf.DefaultPort)
+		}
+		if conf.TestGlobalConfig {
+			SetGlobalConfig(t, snapName, conf.App, conf.DefaultPort)
+		}
+		if conf.TestMixedGlobalAppConfig {
+			SetMixedConfig(t, snapName, conf.App, conf.DefaultPort)
+		}
+	})
+}
+
+// TODO change to TestChangePortLegacyEnv
```

Comment:
Why is the change from `SetEnvConfig` to `TestChangePortLegacyEnv` needed?

Reply:
This is consistent with the other function name changes, to make it clear that:
1) This is executing a test (t.Run)
2) This is testing the changing of port